Account/FirefoxAccount.swift

```diff
@@ -197,67 +198,52 @@ open class FirefoxAccount {
     }
     
     public class FxAProfile {
-        open var avatarUrl: URL?
-        open var avatarImageData: Data?
         open var displayName: String?
         open let email: String
+        open var avatar: Avatar?
         
         init(email: String, displayName: String?, avatar: String?) {
             self.email = email
             self.displayName = displayName
-            self.avatarImageData = nil
 
             if let avatarStr = avatar {
-                self.avatarUrl = URL(string: avatarStr)
-            } else {
-                self.avatarUrl = nil
-            }            
+                self.avatar = Avatar(url: URL(string: avatarStr))
+            }
         }
         
-        func downloadAvatarImage() {
-            guard let avatarUrl = self.avatarUrl else {
-                return
-            }
+        open class Avatar {
+            open var image: UIImage? = nil
+            open var downloadSucceded: Bool? = false
+            open var url: URL?
             
-            DispatchQueue.global().async {
-                self.avatarImageData = try? Data(contentsOf: avatarUrl)
-                NotificationCenter.default.post(name: NotificationFirefoxAccountProfileChanged, object: self)
+            init(url: URL?) {
+                guard let url = url else {
+                    self.image = UIImage(named: "placeholder-avatar")
+                    return
+                }
+                
+                SDWebImageManager.shared().downloadImage(with: url, options: SDWebImageOptions.continueInBackground, progress: nil) { (image, error, cacheType, success, url) in
+                    guard image != nil else {
+                        self.downloadSucceded = false
+                        return
+                    }
+                    self.image = image
+                    NotificationCenter.default.post(name: NotificationFirefoxAccountProfileChanged, object: self)
+                }
             }
         }
     }
     
-    // Fetch current user's FxA profile. It contains the most updated email, displayName and avatar.
+    // Fetch current user's FxA profile. It contains the most updated email, displayName and avatar. This
+    // emits a `NotificationFirefoxAccountProfileChanged` once the profile has finished updating.
```

Comment:
Document that you emit this notification _twice_: once when the profile is updated, and once when its avatar has been loaded from the cache or the network.